apkid/rules/elf/obfuscators.yara

```diff
@@ -372,6 +372,45 @@ rule dexguard_native : obfuscator
       and any of them
 }
 
+rule dexguard_native_arm64 : obfuscator
+{
+  meta:
+    description = "DexGuard (>= 9.0, arm64)"
+    url         = "https://www.guardsquare.com/en/products/dexguard"
+    sample      = "fc3fae3de64eceab969b7d91e3a5fbc45c7407bb8d1a5d5018caa86947604713"
+    author      = "FrenchYeti"
+
+  strings:
+    // that is how dexguard detects frida into /proc/%d/maps
+    $hook = {
+      0b 1d 00 12  //  and        w11,bf,#0xff
+      48 15 40 38  //  ldrb       bf,[x10], #0x1
+      29 25 1b 53  //  ubfiz      w9,w9,#0x5,#0xa
+      29 01 0b 4a  //  eor        w9,w9,w11
+      88 ff ff 35  //  cbnz       bf,LAB_00106e44
+      e8 c1 86 52  //  mov        bf,#0x360f 
+      3f 01 08 6b  //  cmp        w9,bf
+    }
+    // recurring patterns used into several string decryption
+    $str = {
+      6c 69 69 38 //   ldrb       w12,[x11, x9, LSL ]
+      8c ?? ?? 11 //   add        w12,w12,??
+      6c 69 29 38 //   strb       w12,[x11, x9, LSL ]
+      29 05 00 91 //   add        x9,x9,#0x1
+      3f ?? ?? f1 //   cmp        x9,??
+      ec 17 9f 1a //   cset       w12,??
+    }
+    $str2 = {
+      30 ?? cc 9b 10 fe ?? d3 10 a6 0d 9b 6f 69 69 38 d0 69 70 38 
+      0f 02 0f 4a 6f 69 29 38 29 05 00 91 3f ?? ?? f1ef 17 9f 1a
+    }
+    // binaries have always 8 svc instructions
+    $svc = { 01 00 00 d4 }
```

Comment:
Probably we can improve this:


```sh
[0x000016b0]> /ad/ svc.*
0x0000f304             010000d4  svc 0
0x0000f31c             010000d4  svc 0
0x0000f334             010000d4  svc 0
0x0000f34c             010000d4  svc 0
0x0000f364             010000d4  svc 0
0x0000f37c             010000d4  svc 0
0x0000f394             010000d4  svc 0
0x0000f3ac             010000d4  svc 0
```
what about `?? ?? 80 d2` or just adding ``80 d2`?
```sh
[0x000016b0]> pd -1 @@ hit0_*
               0x0000f300      280780d2       mov x8, 0x39             ; '9'
               0x0000f318      080780d2       mov x8, 0x38             ; '8'
               0x0000f330      a80e80d2       mov x8, 0x75             ; 'u'
               0x0000f348      280e80d2       mov x8, 0x71             ; 'q'
               0x0000f360      c80780d2       mov x8, 0x3e             ; '>'
               0x0000f378      c81b80d2       mov x8, 0xde
               0x0000f390      e81a80d2       mov x8, 0xd7
               0x0000f3a8      e80780d2       mov x8, 0x3f             ; '?'
```

Reply:
The same way: 

```sh
[0x000016b0]> pd 2 @@ hit0_*
               ;-- hit0_0:
               0x0000f304      010000d4       svc 0
               0x0000f308      1f0440b1       cmn x0, 1, lsl 12
               ;-- hit0_1:
               0x0000f31c      010000d4       svc 0
               0x0000f320      1f0440b1       cmn x0, 1, lsl 12
               ;-- hit0_2:
               0x0000f334      010000d4       svc 0
               0x0000f338      1f0440b1       cmn x0, 1, lsl 12
               ;-- hit0_3:
               0x0000f34c      010000d4       svc 0
               0x0000f350      1f0440b1       cmn x0, 1, lsl 12
               ;-- hit0_4:
               0x0000f364      010000d4       svc 0
               0x0000f368      1f0440b1       cmn x0, 1, lsl 12
               ;-- hit0_5:
               0x0000f37c      010000d4       svc 0
               0x0000f380      1f0440b1       cmn x0, 1, lsl 12
               ;-- hit0_6:
               0x0000f394      010000d4       svc 0
               0x0000f398      1f0440b1       cmn x0, 1, lsl 12
               ;-- hit0_7:
               0x0000f3ac      010000d4       svc 0
               0x0000f3b0      1f0440b1       cmn x0, 1, lsl 12
```